storage/innobase/buf/buf0dblwr.cc

```diff
@@ -385,8 +386,13 @@ void buf_dblwr_t::recover()
     fil_space_t *space= fil_space_t::get(space_id);
 
     if (!space)
-      /* The tablespace that this page once belonged to does not exist */
+    {
+      /* The tablespace that this page once belonged to does not
+      exist and validate the lsn for these pages in
+      find_encrypted_page(). */
```

Comment:
The comment could be improved:
>The tablespace that this page once belonged to does not seem to exist. However, this could be an encrypted page in full_crc32 format, to be checked by find_encrypted_page().

In fact, I checked and the `FIL_PAGE_LSN` will be stored in clear in `fil_encrypt_buf_for_full_crc32()`. So, we can and should validate the `FIL_PAGE_LSN` already here and avoid preserving invalid pages for `find_encrypted_page()`. In this way, `find_encrypted_page()` will be able to simply invoke `buf_page_is_corrupted(true, …)` without any fear of a false alarm regarding doublewrite buffer pages.